docs/md/executor.md

```diff
@@ -0,0 +1,163 @@
+# Executor
+The workload `Executor` models an intermediate stage of a processing pipeline.
+In a typical workload configuration, an `Executor` uses a supported
+service in the cloud to execute workflows.
+
+## User Guide
+You can configure the type of `Executor` used in your workload by changing the
+`executor` attribute of your workload request.
+
+### `Terra` Executor
+You can execute workflows in a Terra workspace using the `Terra` executor.
+
+The `Terra` executor will...
+
+- Coerce available outputs from an upstream `Source` to a data type acceptable
+  for submission creation
+  (i.e. import a snapshot to the `executor` workspace as a reference)
+- Update the method configuration with the coerced object as its root entity
+  type
+- Launch a submission 
+- Periodically update the statuses of eligible workflows to enable a
+  downstream `Sink` to consume their outputs
+  
+A typical `Terra` executor configuration in the workload request looks like:
+```json
+{
+  "name": "Terra",
+  "workspace": "{workspace-namespace}/{workspace-name}",
+  "methodConfiguration": "{method-configuration-namespace}/{method-configuration-name}",
+  "methodConfigurationVersion": 1,
+  "fromSource": "importSnapshot"
+}
+``` 
+
+And a real-life example for a known method configuration:
+```json
+{
+  "name": "Terra",
+  "workspace": "wfl-dev/CDC_Viral_Sequencing",
+  "methodConfiguration": "wfl-dev/sarscov2_illumina_full",
+  "methodConfigurationVersion": 1,
+  "fromSource": "importSnapshot"
+}
+```
+
+![](./assets/terra-method-configuration.png)
```

Comment:
To satisfy my preference for a concrete example, I found a concrete request with a Terra screenshot provided enough specificity without referencing back to it throughout the doc.